contracts/staking/TokenStaking.sol

```diff
@@ -1463,8 +1531,11 @@ contract TokenStaking is Ownable, IStaking, Checkpoints {
                 index++;
             }
         }
-        for (index = newLength; index < length; index++) {
-            operatorStruct.authorizedApplications.pop();
+
+        address[] storage authorizedApplications = operatorStruct
+            .authorizedApplications;
+        assembly {
+            sstore(authorizedApplications.slot, newLength)
         }
```

Comment:
I don't think it's explained somewhere. You're right, no real cleaning, but in case of adding new element it will rewrite anyway. If you think it's too hacky - I'll change it back

Reply:
I see it for the first time and I feel a little itchy about it 😆 We should not have that many applications in that array so if you ask me, I feel more comfortable with the previous version.

Reply:
Ok, no problem, changed back. We can consider such thing as emergency way to get 90 bytes more 
✔️